presto-main/src/main/java/com/facebook/presto/sql/planner/optimizations/DetermineJoinDistributionType.java

```diff
@@ -224,6 +230,13 @@ private boolean isSmall(PlanNode node)
             return !dataSize.isValueUnknown() && dataSize.getValue() < smallSizeLimit;
         }
 
+        private boolean isMuchSmaller(PlanNode node, PlanNode other)
+        {
+            Estimate nodeDataSize = getOutputSizeEstimate(node);
+            Estimate otherDataSize = getOutputSizeEstimate(other);
+            return !nodeDataSize.isValueUnknown() && !otherDataSize.isValueUnknown() && nodeDataSize.getValue() * (nodeCount - 1) < otherDataSize.getValue();
```

Comment:
-1 isn't a constant, it's the formula for the network cost.  But I'll change it to n to account for the estimation with the cpu cost.